src/test/run-make-fulldeps/coverage-reports-base/expected_show_coverage.async.txt

```diff
@@ -0,0 +1,66 @@
+    1|       |#![allow(unused_assignments)]
+    2|       |
+    3|       |// require-rust-edition-2018
+    4|       |
+    5|      1|async fn f() -> u8 { 1 }
+    6|       |
+    7|       |async fn foo() -> [bool; 10] { [false; 10] }
+    8|       |
+    9|       |pub async fn g(x: u8) {
+   10|       |    match x {
+   11|       |        y if f().await == y => (),
+   12|       |        _ => (),
+   13|       |    }
+   14|      1|}
+   15|       |
+   16|       |// #78366: check the reference to the binding is recorded even if the binding is not autorefed
+   17|       |
+   18|       |async fn h(x: usize) {
+   19|       |    match x {
+   20|       |        y if foo().await[y] => (),
+   21|       |        _ => (),
+   22|       |    }
+   23|      1|}
+   24|       |
+   25|      1|async fn i(x: u8) {
+   26|      1|    match x {
+   27|      1|        y if f().await == y + 1 => (),
+                      ^0                         ^0
```

Comment:
Does the `y` being uncovered also happen in non-async code? I think I remember you mentioning something like this. It's not wrong if you consider that it's the binding of `y`. It is a bit unintuitive, though.

Reply:
Yes, this is the same for async and non-async.

Yes it was discussed because it is less intuitive. At one point, I tried to include the `y` in the pattern code region, but the logic to allow that (which may seem more intuitive, but is counter to what the MIR says) caused problems handling other cases.

I decided it was better to honor the MIR in all cases and show what's really happening.

This is not *just* one of those situations where we can be accurate but non-intuitive, or intuitive but inaccurate; the other problem is, it would require special case handling in the coverage code (which is something we both prefer to avoid). 😄

Reply:
Yeah, special casing is not great. We can always change the way MIR building emits source info at some point.